src/types/firebase/types.ts

```diff
@@ -1,4 +1,23 @@
 import { FirebaseAuthTypes } from '@react-native-firebase/auth';
+import { FirebaseFirestoreTypes } from '@react-native-firebase/firestore';
+
+import FieldValue = FirebaseFirestoreTypes.FieldValue;
 
 export interface FirebaseError extends FirebaseAuthTypes.NativeFirebaseAuthError {}
+
 export interface FirebaseUser extends FirebaseAuthTypes.User {}
+
+export interface FirebaseDocument extends FirebaseFirestoreTypes.DocumentData {}
+
+export interface FirebaseDocumentReference extends FirebaseFirestoreTypes.DocumentReference {}
```

Comment:
you can create type aliases

```
export type FirebaseDocumentReference = FirebaseFirestoreTypes.DocumentReference
```
it would be much more clean :)